test/index.html

```diff
@@ -1,55 +1,153 @@
+      </ol>
+    </section>
+    <section>
+      <h2>Lists: All children of &lt;ul&gt; must only be &lt;li&gt;</h2>
+      <ol>
+        <div>Hello, I'm a div inside a &lt;ul&gt;</div>
+      </ol>
+    </section>
+    <section>
+      <h2>Navigation: Only a single navigation element should exist</h2>
+      <nav>This is the first navigation</nav>
+      <nav>This is the second navigation</nav>
+    </section>
+    <section>
+      <h2>HTML: html tag should have a lang attribute</h2>
+    </section>
+    <section>
+      <h2>Description list: Direct children of a &lt;dl&gt; can only be &lt;dt&gt; and &lt;dd&gt;</h2>
+      <dl>
+        <dt>Hello, I'm dt element inside dl element</dt>
+        <dd>Hello, I'm dd element inside dl element</dd>
+        <p>Hello, I'm p(paragraph) element inside dl element. </p>
+      </dl>
+    </section>
+    <section>
+      <h2>External Links: Provide advanced warning to a user when a link opens in a new window or tab</h2>
+      <a href="https://github.com/jackdomleo7/Checka11y.css" target="_blank">Checka11y Repo</a>
+    </section>
+    <section>
+      <h2>Links: &lt;a&gt; can not be a child of &lt;a&gt;</h2>
+      <a href="https://github.com/jackdomleo7/">
+        <a href="https://github.com/jackdomleo7/Checka11y.css">Hello, I am an anchor tag inside an anchor tag</a>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;audio&gt; with controls attribute can not be a child of &lt;a&gt;</h2>
+      <a>
+        <audio controls></audio>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;button&gt; can not be a child of &lt;a&gt;</h2>
+      <a>
+        <button>Hello, I am a button inside an &lt;a&gt;</button>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;details&gt; can not be a child of &lt;a&gt;</h2>
+      <a>
+          <details>Hello, I am a details inside an anchor tag</details>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;embed&gt; can not be a child of &lt;a&gt;</h2>
+      <a>
+        <embed src="./assets/a11y.png" />
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;iframe&gt; can not be a child of &lt;a&gt;</h2>
+      <a>
+        <iframe src="https://jackdomleo.dev" height="500" width="500" title="test for anchor tag with iframe as a child"></iframe>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;img&gt; with usemap attribute can not be a child of &lt;a&gt;</h2>
+      <a>
+        <img src="./assets/a11y.png" alt="test for anchor tag with usemap attribute" usemap/>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;input&gt; without type hidden attribute can not be a child of &lt;a&gt;</h2>
+      <a>
+          <input type="text" placeholder="I am an input tag"></input>
+      </a>
+    </section>
+    <section>
+      <h2>Links: &lt;keygen&gt; can not be a child of &lt;a&gt;</h2>
+      <a>
+          <keygen name="second" />
```

Comment:
@danspratling Can I get your opinion on something? `keygen` was removed in HTML5 (along with many other HTML elements), so should we make this a HTML5 only tool, or should we still support for non HTML5 elements? We could raise an issue for a new feature to check for all deprecated HTML elements? 🤔 @wulforr feel free to add your thoughts here too 😊

Reply:
I would remove all depricated code. We could get into a huge rabbit hole if we try to support everything which has already been removed.

Compatibility checks are outside of the scope of these tests HOWEVER if we do decide to do anything with outdated code it should only be to suggest it is removed.

Reply:
I agree with @danspratling we should remove depricated code.
Other than that we can also remove anchor tag inside anchor tag as browser automatically corrects it.[More info here](https://stackoverflow.com/a/13054959/11573007)

Reply:
Alright, that sounds good, @wulforr, would you be able to remove any deprecated HTML elements that aren't in HTML5 and I'll update the project and README to reflect it can be a HTML5 validator. Sorry, I realise this wasn't in scope of your original work - would this be ok?

Reply:
Yeah sure, no problem